Scripts/DowloadPortalItemData.swift

```diff
@@ -126,61 +126,47 @@ func uncompressArchive(at sourceURL: URL, to destinationURL: URL) throws {
+/// Downloads a file from a given portal and writes it to a given path.
 /// - Parameters:
 ///   - sourceURL: The portal URL to the resource.
-///   - downloadDirectory: The directory that stores downloaded data.
-///   - completion: A closure to handle the results.
-func downloadFile(at sourceURL: URL, to downloadDirectory: URL, completion: @escaping (Result<URL, Error>) -> Void) {
-    let downloadTaskCompleted = { (temporaryURL: URL?, response: URLResponse?, error: Error?) in
-        if let temporaryURL = temporaryURL,
-           let response = response,
-           let suggestedFilename = response.suggestedFilename {
-            do {
-                let downloadName: String
-                let isArchive = (suggestedFilename as NSString).pathExtension == "zip"
-                // If the downloaded file is an archive and contains
-                //   - 1 file, use the name of that file.
-                //   - multiple files, use the suggested filename (*.zip).
-                // If it is not an archive, use the server suggested filename.
-                if isArchive {
-                    let count = try count(ofFilesInArchiveAt: temporaryURL)
-                    if count == 1 {
-                        downloadName = try name(ofFileInArchiveAt: temporaryURL)
-                    } else {
-                        downloadName = suggestedFilename
-                    }
-                } else {
-                    downloadName = suggestedFilename
-                }
-                
-                let downloadURL = downloadDirectory.appendingPathComponent(downloadName, isDirectory: false)
-                
-                if FileManager.default.fileExists(atPath: downloadURL.path) {
-                    try FileManager.default.removeItem(at: downloadURL)
-                }
-                
-                if isArchive {
-                    let extractURL = downloadURL.pathExtension == "zip"
-                        // Uncompresses to directory named after archive.
-                        ? downloadURL.deletingPathExtension()
-                        // Uncompresses to appropriate subdirectory.
-                        : downloadURL.deletingLastPathComponent()
-                    try uncompressArchive(at: temporaryURL, to: extractURL)
-                } else {
-                    try FileManager.default.moveItem(at: temporaryURL, to: downloadURL)
-                }
-                
-                completion(.success(downloadURL))
-            } catch {
-                completion(.failure(error))
-            }
-        } else if let error = error {
-            completion(.failure(error))
+///   - downloadDirectory: The directory to store the downloaded data in.
+/// - Throws: Exceptions when downloading, naming, uncompressing, and moving the file.
+/// - Returns: The name of the downloaded file.
+func downloadFile(from sourceURL: URL, to downloadDirectory: URL) async throws -> String? {
```

Comment:
It will not have a filename if there isn't a `suggestedFilename` returned from the server. This replicates the previous scripts behavior. See [this comment](https://github.com/Esri/arcgis-maps-sdk-swift-samples/pull/361#discussion_r1543525873) from when these changes were originally part of another PR.